lib/use-checkout-pricing.js

```diff
@@ -38,20 +47,39 @@ export default function useCheckoutPricing(initialInputs = {}, handleError = thr
   const recurly = useRecurly();
   const [loading, setLoading] = useState(false);
   const [input, setInput] = useState(initialInputs);
-  let checkoutPricing = recurly.Pricing.Checkout();
-  const [pricing, setPricing] = useState(checkoutPricing);
+  const [pricing, setPricing] = useState(recurly.Pricing.Checkout());
 
   useEffect(() => {
     setLoading(true);
-    const { subscriptions = [] } = input;
+    let checkoutPricing = recurly.Pricing.Checkout();
+    const { subscriptions = [], ...singleInputs } = input;
+
+    checkoutPricing = addSingleInputs(singleInputs, checkoutPricing);
+
     addSubscriptions(subscriptions, checkoutPricing)
       .then(() => {
-        setPricing(checkoutPricing);
+        checkoutPricing = checkoutPricing.reprice().done(() => {
+          setPricing(checkoutPricing);
+          setLoading(false);
+        });
       })
-      .catch(handleError)
-      .finally(() => setLoading(false));
+      .catch(error => {
+        handleError(error);
+        setLoading(false);
+      });
   }, [input]);
 
+  function addSingleInputs(singleInputs, checkoutPricing) {
+    const { PRICING_METHODS } = checkoutPricing;
+    const pricingMethodsWhiteList = PRICING_METHODS.filter(str => !['reset', 'remove', 'reprice'].includes(str));
+    const whiteListedInputs = pick(singleInputs, pricingMethodsWhiteList);
```

Comment:
Wondering if we can remove the `pick` dependency by whitelisting like this:

```js
const { PRICING_METHODS } = checkoutPricing;
const permittedInputs = PRICING_METHODS
  .filter(method => !['reset', 'remove', 'reprice'].includes(method))
  .reduce((acc, method) => (acc[method] = singleInputs[method], acc), {});
```

Reply:
https://github.com/recurly/react-recurly/pull/25/commits/13f2b766bfdc59884e9dda3334be10020317a3b1